frame/staking/src/lib.rs

```diff
@@ -2106,7 +2115,15 @@ impl<T: Config> Module<T> {
 					l.active += amount;
 					l.total += amount;
 					let r = T::Currency::deposit_into_existing(stash, amount).ok();
-					Self::update_ledger(&controller, &l);
+					// Best effort we try to update the ledger... but if it fails, not much we can do.
+					// Not a major issue since we did give the account the funds it wanted, just weren't
+					// able to re-stake it.
+					//
+					// Although, this should always work because the only thing that can cause the ledger
+					// update to fail is adding too many locks to an account, and this operation should only
+					// update an existing lock.
+					let res = Self::update_ledger(&controller, &l);
+					debug_assert!(res.is_ok(), "failed to update ledger when rewarding to staked");
```

Comment:
```suggestion
					let _res = Self::update_ledger(&controller, &l);
					debug_assert!(_res.is_ok(), "failed to update ledger when rewarding to staked");
```

I think that in a release build, without this change, we'll see a warning about an unused variable.